common/DatabaseController.js

```diff
@@ -0,0 +1,106 @@
+const { collection, doc, getDoc, getDocs, query, where } = require('firebase/firestore');
+const firebase = require('firebase/app');
+const moment = require('moment');
+const firebaseApp =require("firebase/app");
+const firestore = require ("firebase/firestore");
+
+
+class DatabaseController {
+    constructor() {
+        this.firebaseApp_= firebaseApp.initializeApp({
+            apiKey: "AIzaSyAGqShV0l2hfX2eO8hYzgXnXDmpO5F0xsI",
+            authDomain: "silken-tenure-419721.firebaseapp.com",
+            projectId: "silken-tenure-419721",
+            storageBucket: "silken-tenure-419721.appspot.com",
+            messagingSenderId: "245250661585",
+            appId: "1:245250661585:web:194d2c6ccce99dd6aab9f0",
+            measurementId: "G-L0BFNM9LDG"
+        });
+        this.db = firestore.getFirestore("myrestaurant");
+    }
+
+
+    // async getAllMenu() { 
+    //     const dishCol = collection(this.db, 'Dish');
+    //     const dishSnapshot = await getDocs(dishCol);
+    //     const list = dishSnapshot.docs.map(doc => doc.data());
+    //     console.log(list);
+    //     return list;
+    // }
+
+    async getDishById(dishId) {
+      const dishDocRef = doc(this.db, 'Dish', dishId);
+      const dishDoc = await getDoc(dishDocRef);
+      if (dishDoc.exists()) {
+          let dishData = dishDoc.data();
+          return {
+              ...dishData,
+              // Include the document ID if necessary
+              id: dishDoc.id,
+          };
+      } else {
+          return null;
```

Comment:
Better if we handle this like error from here

Reply:
Added